src/main/java/gregtech/api/pattern/MultiblockShapeInfo.java

```diff
@@ -33,14 +37,35 @@ public BlockInfo[][][] getBlocks() {
     }
 
     public static Builder builder() {
-        return new Builder();
+        return new Builder(RIGHT, DOWN, BACK);
+    }
+
+    public static Builder builder(RelativeDirection... structureDir) {
+        return new Builder(structureDir);
     }
 
     public static class Builder {
 
+        private final RelativeDirection[] structureDir = new RelativeDirection[3];
+
         private List<String[]> shape = new ArrayList<>();
         private Map<Character, BlockInfo> symbolMap = new HashMap<>();
 
+        public Builder(RelativeDirection... structureDir) {
```

Comment:
i think that the VarArgs should be checked to make sure it is not null and has exactly three nonnull elements. otherwise you could pass in one or two (or none) directions, which would cause problems. 
once checked, you could pass that in to a new constructor with 
`Builder(RelativeDirection x, RelativeDirection y, RelativeDirection z)` or something. 
maybe it would be ideal to add `@NotNull` to the paramaters as well.